src/Data/Formatter/Parser/Number.purs

```diff
@@ -0,0 +1,61 @@
+module Data.Formatter.Parser.Number
+  ( parseInteger
+  , parseMaybeInteger
+  , parseNumber
+  , parseDigit
+  ) where
+
+import Prelude
+
+import Data.Int (toNumber, floor)
+import Data.Array (some, many, length)
+import Data.Formatter.Parser.Number (parseDigit)
+import Data.Formatter.Internal (foldDigits)
+import Data.Function (on)
+import Data.Tuple (Tuple(..))
+import Text.Parsing.Parser as P
+import Text.Parsing.Parser.Combinators as PC
+import Data.Formatter.Parser.Utils (oneOfAs)
+import Text.Parsing.Parser.String as PS
+import Data.Maybe (Maybe(..))
+import Math as Math
+
+
+parseInteger ∷ ∀ s m. Monad m => PS.StringLike s => P.ParserT s m Int
+parseInteger = some parseDigit <#> foldDigits
+
+parseMaybeInteger ∷ ∀ s m. Monad m => PS.StringLike s => P.ParserT s m (Maybe Int)
+parseMaybeInteger = many parseDigit <#> (\l -> if length l == 0 then Nothing else Just $ foldDigits l)
+
+parseFractional ∷ ∀ s m. Monad m => PS.StringLike s => P.ParserT s m Number
+parseFractional = parseInteger <#> case _ of
+  0 ->  0.0
+  n -> (toNumber n) / (pow 10 $ numOfDigits n)
+
+parseNumber ∷ ∀ s m. Monad m => PS.StringLike s => P.ParserT s m Number
+parseNumber = (+)
+  <$> (parseInteger <#> toNumber)
+  <*> (PC.option 0.0 $ PC.try $ PS.oneOf ['.', ','] *> parseFractional)
+
+pow :: Int -> Int -> Number
+pow = Math.pow `on` toNumber
```

Comment:
hm, so if number of digits are too much`pow(10,numDigit)` would overflow.
in chrome debugger `Math.pow(10,308)` returns `1.0000000000000006e+308`.
`Math.pow(10,309)` returns Infinity.

oh actually I could just prefix it with `"0."` and do parseFloat